packages/compass-schema-validation/src/stores/store.ts

```diff
@@ -0,0 +1,117 @@
+import { editModeChanged } from '../modules/edit-mode';
+import {
+  localAppRegistryActivated,
+  globalAppRegistryActivated,
+} from '@mongodb-js/mongodb-redux-common/app-registry';
+import semver from 'semver';
+import type { CollectionTabPluginMetadata } from '@mongodb-js/compass-collection';
+import type { AppRegistry } from 'hadron-app-registry';
+import type { DataService } from 'mongodb-data-service';
+import type { MongoDBInstance } from '@mongodb-js/compass-app-stores/provider';
+
+/**
+ * The lowest supported version.
+ */
+const MIN_VERSION = '3.2.0';
+
+// Exposed for testing
+export function createBareStore(): Store<RootState, RootAction> {
+  return createStore(reducer, applyMiddleware(thunk));
+}
+
+/**
+ * The store has a combined pipeline reducer plus the thunk middleware.
+ */
+export function onActivated(
+  options: CollectionTabPluginMetadata,
+  {
+    localAppRegistry,
+    globalAppRegistry,
+    dataService,
+    instance,
+  }: {
+    localAppRegistry: AppRegistry;
+    globalAppRegistry: AppRegistry;
+    dataService: DataService;
+    instance: MongoDBInstance;
+  }
+) {
+  const store = createBareStore();
+  const cleanup: (() => void)[] = [];
+  function on(
+    eventEmitter: {
+      on(ev: string, l: (...args: any[]) => void): void;
+      removeListener(ev: string, l: (...args: any[]) => void): void;
+    },
+    ev: string,
+    listener: (...args: any[]) => void
+  ) {
+    eventEmitter.on(ev, listener);
+    cleanup.push(() => eventEmitter.removeListener(ev, listener));
+  }
+
+  // Set the app registry if preset. This must happen first.
+  store.dispatch(localAppRegistryActivated(localAppRegistry));
+  store.dispatch(globalAppRegistryActivated(globalAppRegistry));
+  store.dispatch(serverVersionChanged(instance.build.version));
+  store.dispatch(dataServiceConnected(dataService));
+
+  /**
+   * When the collection is changed, update the store.
+   */
+  on(localAppRegistry, 'fields-changed', (fields) => {
+    store.dispatch(fieldsChanged(fields.fields));
+  });
+
+  const setEditMode = () => {
+    store.dispatch(
+      editModeChanged({
+        collectionTimeSeries: !!options.isTimeSeries,
+        collectionReadOnly: options.isReadonly ? true : false,
+        writeStateStoreReadOnly: !instance.isWritable,
+        oldServerReadOnly: semver.gte(MIN_VERSION, instance.build.version),
+      })
+    );
+  };
+
+  // set the initial value
+  setEditMode();
+
+  // isWritable can change later
+  instance.on('change:isWritable', () => {
+    store.dispatch(
+      editModeChanged({
+        writeStateStoreReadOnly: !instance.isWritable,
+      })
+    );
+  });
+
+  if (options.namespace) {
+    const namespace = toNS(options.namespace);
+    store.dispatch(namespaceChanged(namespace));
+  }
+
+  // XXX: Verify that this is the correct way of handling this now that tabs are loaded ad-hoc?
+  (store.dispatch as ThunkDispatch<RootState, unknown, RootAction>)(
+    activateValidation()
+  );
```

Comment:
@gribnoysup See the comment, this seems reasonable as a dispatched action, do we just call it in the `activate()` method?

Reply:
I don't see anything wrong with that. If it's "sync" state, you can pass it to the configure store, if it's async action that needs to run when the plugin is first activated, you can dispatch it in the activate method